src/notifications/ns_dialog.py

```diff
@@ -57,3 +53,17 @@ def hide(self):
     def mousePressEvent(self, event):
         if event.button() == QtCore.Qt.RightButton:
             self.hide()
+
+    def updatePosition(self):
+        screen = QtGui.QDesktopWidget().screenGeometry()
+        dialog_size = self.geometry()
+        position = self.settings.popup_position  # self.client.notificationSystem.settings.popup_position
```

Comment:
Oh, right, to express this in ZeP-insanity-parent-pointer-language. Blarg. More restructuring needed. :P